spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -68,14 +68,18 @@ def create_call(user_params)
   end
 
   describe 'PUT #update' do
-    def update_user_call(user_params)
+    def update_user_call(user_params, headers)
+      request.headers.merge! headers
       put :update, params: { id: user.id, user: user_params }
     end
 
     context 'when is successful' do
       let(:user_params) { { email: 'email@dominio.com', password: '123456' } }
+      let(:headers) do
+        { 'Authorization': JsonWebToken.encode(user_id: user.id) }
+      end
```

Comment:
Maybe you can create a factory for it in the users factory like `authenticated_user`

Reply:
Do you mean a factory that returns header only? Or does returns the user with the token?

Reply:
No it should add the headers to the request and also return an user logged in.